crates/pop-parachains/src/build/runtime.rs

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::Error;
+use duct::cmd;
+use pop_common::Profile;
+pub use srtool_lib::{get_image_digest, get_image_tag, ContainerEngine};
+use std::{env, fs, path::PathBuf};
+
+const DEFAULT_IMAGE: &str = "docker.io/paritytech/srtool";
+const ONE_HOUR: u64 = 60 * 60;
+
+/// Builds and executes the command for running the deterministic runtime build process using
+/// srtool.
+pub struct SrToolBuilder {
+	cache_mount: String,
+	default_features: String,
+	digest: String,
+	engine: ContainerEngine,
+	image: String,
+	package: String,
+	path: PathBuf,
+	profile: Profile,
+	runtime_dir: PathBuf,
+	tag: String,
+}
+
+impl SrToolBuilder {
+	/// Creates a new instance of `SrToolBuilder`.
+	///
+	/// # Arguments
+	/// * `engine` - The container engine to use.
+	/// * `path` - The path to the project.
+	/// * `package` - The runtime package name.
+	/// * `profile` - The profile to build the runtime.
+	/// * `runtime_dir` - The directory path where the runtime is located.
+	pub fn new(
+		engine: ContainerEngine,
+		path: Option<PathBuf>,
+		package: String,
+		profile: Profile,
+		runtime_dir: PathBuf,
+	) -> Result<Self, Error> {
+		let default_features = String::new();
+		let tag = get_image_tag(Some(ONE_HOUR)).map_err(|_| Error::ImageTagRetrievalFailed)?;
```

Comment:
```suggestion
		let tag = get_image_tag(Some(TIMEOUT)).map_err(|_| Error::ImageTagRetrievalFailed)?;
```
Assume ONE_HOUR is the timeout?

Reply:
https://github.com/r0gue-io/pop-cli/pull/425/commits/c7dea912f7efe5ac15c68d47e13e6dfc6c2b85cc